src/valid/interface.rs

```diff
@@ -215,6 +215,11 @@ impl VaryingContext<'_> {
                 if !needs_interpolation && interpolation.is_some() {
```

Comment:
`needs_interpolation` has to change now, since WGSL demands the vertex outputs to have matching interpolation, and we try to follow WGSL in the IR

Reply:
Okay, so this entails finding the matching declaration in the next / previous pipeline stage and checking them against each other, right?

Reply:
This code doesn't know about which shaders this will connect to. It's going to be `wgpu` that validates the pipeline interface. But this code can at least check that vertex outputs and fragment inputs *have* interpolation. Right now, `needs_interpolation` is only true for fragment inputs

Reply:
Ah, okay. That makes more sense now.